test/js/containers/FormPage.unit.spec.jsx

```diff
@@ -196,6 +196,51 @@ describe('Schemaform <FormPage>', () => {
 
     expect(router.push.calledWith('first-page'));
   });
+  it('should not show a Back button on the first page', () => {
+    const route = {
+      pageConfig: {
+        pageKey: 'testPage',
+        schema: {},
+        uiSchema: {},
+        errorMessages: {},
+        title: ''
+      },
+      pageList: [
+        {
+          path: 'testing',
+          pageKey: 'testPage'
+        }
+      ]
+    };
+    const form = {
+      pages: {
+        testPage: {
+          depends: () => false,
+          schema: {},
+          uiSchema: {},
+        }
+      },
+      data: {}
+    };
+    const user = {
+      profile: {
+        savedForms: []
+      },
+      login: {
+        currentlyLoggedIn: false
+      }
+    };
+
+    const tree = SkinDeep.shallowRender(
+      <FormPage
+        form={form}
+        user={user}
+        route={route}
+        location={location}/>
```

Comment:
I think the reason why you're seeing the current location path not match the route is because the `location` here is set at the top of this file to be `testing/0`

Reply:
If you change this to be:
```js
...
location={{pathname: 'testing'}}/>
...
```
the test passes

Reply:
Yeah, at the top of the file it's set as `location: { pathname: '/testing/0' } }`. I think this is supposed to match the browser's `window.location.pathname` property which is what the router is looking at. Specifically it should have the leading `/` which it does. The problem is that the `route.pageList.path` *does not* have the leading slash so they don't equal. 

Also it has that extra `/0`. I don't know why it is there. It doesn't seem to make any of the tests fail if it is removed, so I can certainly remove that.

I think this may just be a question of getting the router into the right state. Looking at the starter app, it has a [urlPrefix](https://github.com/usds/us-forms-system-starter-app/blob/master/js/config/form.js#L10) that just happens to be `/` by default, so it should work there. It just seems odd that the `path` and `pathname` don't match here. Although I can set the data with a slash-less `location` to make this pass, I still don't understand why and it feels like cheating.

Also, I don't think it's necessary to create the built files when running against the unit tests. The `npm run test` builds everything using Babel before starting the tests. That's super handy and saves lots of time!

Reply:
It looks like I need to use `formConfig.urlPrefix` in the test, which I've done. If none is set I have it using `'/'` for a default, the test fails if I use `''` and I'm not really sure why. The "it renders" test above expected both the Back and Continue button so I had to change it.

Reply:
To answer a couple of the points of confusion above:

#### When to use '/' or not
In regards to when the prefixed `/` is included or not, I think what happens is the paths are generated from what is specified in the `formConfig`. In the `formConfig`, paths are not written with the prefixed `/`. But in generating the routes, a new object called `pageList` is created, which takes the paths from `formConfig` and adds the `urlPrefix` to them, which in most cases is just a `/`. The router then uses the `pageList` object to determine which is the next and previous pages to go to, and matches that against the `location.pathname`, which always includes the `/` for that path.

For this reason, I think the test code is a bit confusing because we're handwriting the `pageList` object, which is actually generated by the library code. For some reason, the existing test code in this file doesn't include the `/` for the paths in `pageList`, but the actual code would, assuming `urlPrefix` is passed to the `formConfig`. For that reason I think it's a mistake to manually add `urlPrefix` to the path in `pageList` when checking against `location.pathname`, because that should already be included.

#### About the `/0` in the original `location.pathname`
I _think_ this was included because some of the tests are testing array functionality, in which cases items in the array are indexed, and the route tells us which item we're currently on. It's strange that removing this doesn't break the tests, which I think tells me that the test code isn't written accurately in regards to paths. But I need to investigate array paths more...